source/util/migration_handler.ts

```diff
@@ -1,12 +1,12 @@
+import { Client } from 'pg'
 import { migrate } from 'postgres-migrations'
 
 export async function applyMigrations(config: any): Promise<void> {
-	const dbConfig = {
-		database: config['database'] || 'postgres',
-		user: config['user'] || 'postgres',
-		password: config['password'] || 'postgres',
-		host: config['host'] || 'localhost',
-		port: config['port'] || 5432,
+	const client = new Client(config)
+	await client.connect()
+	try {
+		await migrate({ client }, __dirname + '/migrations')
+	} finally {
+		await client.end()
```

Comment:
`postgres-migrations` requires a connection to already be established when using `pg.Client`. It also will not close the connection for you.

https://github.com/thomwright/postgres-migrations?tab=readme-ov-file#api